bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/handler/IpCameraHandler.java

```diff
@@ -0,0 +1,1795 @@
+                }
+                if (!cameraConfig.getUser().isEmpty()) {
+                    filterOptions += " ";// add space as the Framework does not allow spaces at start of config.
+                }
+                ffmpegRtspHelper = new Ffmpeg(this, format, cameraConfig.getFfmpegLocation(), inputOptions, input,
+                        filterOptions + cameraConfig.getMotionOptions(), OutputOptions, username, password);
+                ffmpegRtspHelper.startConverting();
+                break;
+            case MJPEG:
+                if (ffmpegMjpeg == null) {
+                    if (inputOptions.isEmpty()) {
+                        inputOptions = "-hide_banner -loglevel warning";
+                    } else {
+                        inputOptions = inputOptions + " -hide_banner -loglevel warning";
+                    }
+                    ffmpegMjpeg = new Ffmpeg(this, format, cameraConfig.getFfmpegLocation(), inputOptions, rtspUri,
+                            cameraConfig.getMjpegOptions(), "http://127.0.0.1:" + serverPort + "/ipcamera.jpg",
+                            username, password);
+                }
+                if (ffmpegMjpeg != null) {
+                    ffmpegMjpeg.startConverting();
+                }
+                break;
+            case SNAPSHOT:
+                // if mjpeg stream you can use ffmpeg -i input.h264 -codec:v copy -bsf:v mjpeg2jpeg output%03d.jpg
+                if (ffmpegSnapshot == null) {
+                    if (inputOptions.isEmpty()) {
+                        // iFrames only
+                        inputOptions = "-threads 1 -skip_frame nokey -hide_banner -loglevel warning";
+                    } else {
+                        inputOptions = inputOptions + " -threads 1 -skip_frame nokey -hide_banner -loglevel warning";
+                    }
+                    ffmpegSnapshot = new Ffmpeg(this, format, cameraConfig.getFfmpegLocation(), inputOptions, rtspUri,
+                            "-an -vsync vfr -update 1", "http://127.0.0.1:" + serverPort + "/snapshot.jpg", username,
+                            password);
+                }
+                if (ffmpegSnapshot != null) {
+                    ffmpegSnapshot.startConverting();
+                }
+                break;
+        }
+    }
+
+    public void noMotionDetected(String thisAlarmsChannel) {
+        setChannelState(thisAlarmsChannel, OnOffType.OFF);
+        firstMotionAlarm = false;
+        motionAlarmUpdateSnapshot = false;
+        motionDetected = false;
+        if (streamingAutoFps) {
+            stopSnapshotPolling();
+        } else if (updateImageWhen.contains("4")) { // During Motion Alarms
+            stopSnapshotPolling();
+        }
+    }
+
+    // Change alarms that are not counted as motion detecting.
+    public void changeAlarmState(String thisAlarmsChannel, String state) {
+        updateState(thisAlarmsChannel, OnOffType.valueOf(state));
+    }
```

Comment:
This method is misleading since it affects any channel given (not just an alarm channel). Also since you are expecting either `ON` or `OFF` as the state value you might as well enforce by changing the `state` parameter to an `OnOffType` instead of String.

Reply:
Made the change to using OnOffType as suggested and also added javadoc to make things clear and easier to check the intension by hovering over the method.